src/providers/data_provider/dp_request.c

```diff
@@ -396,6 +399,9 @@ static void dp_req_done(struct tevent_req *subreq)
 
     DP_REQ_DEBUG(SSSDBG_TRACE_FUNC, state->dp_req->name,
                  "Request handler finished [%d]: %s", ret, sss_strerror(ret));
+    DP_REQ_DEBUG(SSSDBG_PERF_STAT, state->dp_req->name,
+                 "Handling request took [%"PRIu64"] microseconds",
```

Comment:
(here and in `sdap_call_op_callback()`)

Do we really need time measured in microseconds?  I'd say milliseconds precision would be more than enough.

Any in any case I think it would be more "user friendly" to change representation to "seconds:milliseconds[:microseconds]" (or other delimiter)

Reply:
For this purpose storing `timeval` and using `tevent_timeval_until()` might be handy.

Reply:
I would prefer to use the same unit for all timing values. I agree that ms might be sufficient for whole requests but I think it would be confusing to mix ms and us.

Reply:
To mix ms and us where?

Reply:
This representation - `[%"PRIu64"] microseconds` - is going to be really unreadable, imo.

Reply:
I use milliseconds in the latest version with 3 digits after the separator to still have microsecond resolution because many LDAP operations are in this order. I moved the generation of the string to a separate function to make it easier to use a common format and make changes more easy as well.

Reply:
TBH, I don't think LDAP network operation under 1ms is currently realistic.

But more important, this depends on how you see the purpose of those patches.

My understanding is it is "warning of high queries time". In this respect, 1-2 decimal point after seconds is more than enough and much more user friendly.

Reply:
> But more important, this depends on how you see the purpose of those patches.

I mean, I can't imagine a scenario where one needs to debug execution time of LDAP network requests with microseconds precision.

Imo, motivation behind RFE was to catch cases where SSSD hits a timeout (that is measured in seconds).

This is a typical output where SSSD fetches a lot of data (i.e. exactly the case that might require attention):
```
Handling LDAP operation [1][server: [...:389] filter: [(objectclass=*)] base: []] took [127.106] milliseconds.
Handling LDAP operation [2][server: [...:389] filter: [...] base: [dc=example,dc=test]] took [125.591] milliseconds.
Handling LDAP operation [3][server: [...:389] filter: [(&(uid=doo2)(objectclass=posixAccount)(uid=*)(&(uidNumber=*)(!(uidNumber=0))))] base: [dc=example,dc=test]] took [126.038] milliseconds.
DP Request [Account #5]: Handling request took [157.645] milliseconds.
Handling LDAP operation [4][server: [...:389] filter: [(&(gidNumber=1456412)(objectClass=posixGroup)(cn=*)(&(gidNumber=*)(!(gidNumber=0))))] base: [dc=example,dc=test]] took [1171.351] milliseconds.
Handling LDAP operation [5][server: [...:389] filter: [(null)] base: [cn=ldapusers_2,ou=Groups,dc=example,dc=test]] took [6448.194] milliseconds.
DP Request [Account #6]: Handling request took [57262.792] milliseconds.
Handling LDAP operation [6][server: [...:389] filter: [(&(uid=doo2)(objectclass=posixAccount)(&(uidNumber=*)(!(uidNumber=0))))] base: [dc=example,dc=test]] took [130.368] milliseconds.
Handling LDAP operation [7][server: [...:389] filter: [(&(uniquemember=uid=doo2,ou=People,dc=example,dc=test)(objectClass=posixGroup)(cn=*))] base: [dc=example,dc=test]] took [127.258] milliseconds.
Handling LDAP operation [8][server: [...:389] filter: [(&(uniquemember=cn=ldapusers_2,ou=Groups,dc=example,dc=test)(objectClass=posixGroup)(cn=*))] base: [dc=example,dc=test]] took [125.393] milliseconds.
Handling LDAP operation [9][server: [...:389] filter: [(&(uniquemember=cn=ldapusers_1,ou=Groups,dc=example,dc=test)(objectClass=posixGroup)(cn=*))] base: [dc=example,dc=test]] took [124.135] milliseconds.
DP Request [Initgroups #7]: Handling request took [14719.387] milliseconds.
Handling LDAP operation [10][server: [...:389] filter: [(&(gidNumber=1456411)(objectClass=posixGroup)(cn=*)(&(gidNumber=*)(!(gidNumber=0))))] base: [dc=example,dc=test]] took [129.820] milliseconds.
Handling LDAP operation [11][server: [...:389] filter: [(objectclass=posixAccount)] base: [uid=doo1,ou=People,dc=example,dc=test]] took [125.941] milliseconds.
DP Request [Account #8]: Handling request took [16231.411] milliseconds.
```

Well, `57262.792` isn't _that_ unreadable, after all... If you strongly prefer to keep it, I won't object.